lib/apiservers/engine/backends/cache/container_cache.go

```diff
@@ -129,3 +130,25 @@ func (cc *CCache) GetContainerFromExec(eid string) *container.VicContainer {
 	}
 	return nil
 }
+
+func (cc *CCache) UpdateContainerName(oldName, newName string) error {
+	cc.m.Lock()
+	defer cc.m.Unlock()
+
+	container := cc.getContainer(oldName)
+	if container == nil {
+		return fmt.Errorf("no such container: %s", oldName)
+	}
+
+	delete(cc.containersByName, container.Name)
+	if exists := cc.getContainer(newName); exists != nil {
```

Comment:
If you check if the new name is available AFTER deleting the container reference from the name list, will this affect being able to lookup by name?